File: test-suite/src/main/java/example/micronaut/agentic/RecipeAdvisorAgent.java
package example.micronaut.agentic;

import dev.langchain4j.agentic.Agent;
import dev.langchain4j.service.UserMessage;
import dev.langchain4j.service.V;
import io.micronaut.langchain4j.agentic.annotation.AgenticService;

/**
 * Suggests a popular recipe for a given country.
 * Derived agent id: "recipe-advisor".
 */
@AgenticService(outputName = "recipe")
public interface RecipeAdvisorAgent {

    @UserMessage("""
        You are a cooking expert.
        Suggest a popular recipe in {{country}}.
        Reply with the country name and the recipe name separated by a minus sign, then the details.
        """)
    @Agent(description = "Suggests a popular recipe based on a user request")
    String suggestRecipe(@V("country") String country);
}

File: test-suite-python/src/main/python/example/micronaut/agentic/RecipeAdvisorAgent.py
from abc import ABC, abstractmethod
from typing import Annotated

from dev.langchain4j.agentic import Agent
from dev.langchain4j.service import UserMessage, V
from micronaut.langchain4j.agentic.annotation import AgenticService


# Suggests a popular recipe for a given country.
# Derived agent id: "recipe-advisor".
@AgenticService(outputKey="recipe")
class RecipeAdvisorAgent(ABC):

    @UserMessage("""
        You are a cooking expert.
        Suggest a popular recipe in {{country}}.
        Reply with the country name and the recipe name separated by a minus sign, then the details.
        """)
    @Agent(description="Suggests a popular recipe based on a user request")
    @abstractmethod
    def suggest_recipe(self, country: Annotated[str, V("country")]) -> str:
        ...
